### PR TITLE
Add SBRPs to enable PVP flow for arcade repo

### PR DESCRIPTION
**Pull request by mthalman**
When attempting to enable PVP flow for the arcade repo, it exposed its dependencies on several packages that need to be provided in SBRP:

* Microsoft.CodeAnalysis.CSharp.2.9.0
  * New dependencies:
    * Microsoft.CodeAnalysis.Common.2.9.0
    * System.Diagnostics.StackTrace.4.3.0
    * System.Text.Encoding.CodePages.4.3.0
    * System.Threading.Tasks.Parallel.4.3.0
    * System.ValueTuple.4.3.0
    * System.Xml.XPath.4.3.0
    * System.Xml.XPath.XDocument.4.3.0
* Microsoft.Extensions.DependencyInjection.6.0.0
  * New dependencies:
    * Microsoft.Extensions.DependencyInjection.Abstractions.6.0.0
* Microsoft.Extensions.DependencyModel.2.1.0
  * New dependencies:
    * Microsoft.DotNet.PlatformAbstractions.2.1.0
* Microsoft.Extensions.Logging.Console.6.0.0
  * New dependencies:
    * Microsoft.Extensions.Configuration.6.0.0
    * Microsoft.Extensions.Configuration.Abstractions.6.0.0
    * Microsoft.Extensions.Configuration.Binder.6.0.0
    * Microsoft.Extensions.Logging.6.0.0
    * Microsoft.Extensions.Logging.Abstractions.6.0.0
    * Microsoft.Extensions.Logging.Configuration.6.0.0
    * Microsoft.Extensions.Logging.Console.6.0.0
    * Microsoft.Extensions.Options.6.0.0
    * Microsoft.Extensions.Options.ConfigurationExtensions.6.0.0
    * Microsoft.Extensions.Primitives.6.0.0
    * System.Diagnostics.DiagnosticSource.6.0.0
* System.IO.Packaging.4.5.0

Manual changes needed to be made to the Microsoft.Extensions.Primitives generated code due to https://github.com/dotnet/sdk/issues/31570.

Contributes to https://github.com/dotnet/source-build/issues/3043

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@mthalman - looks like there is a missing dependency, Newtonsoft.Json.9.0.1, coming from https://github.com/dotnet/source-build-reference-packages/pull/595/files#diff-feaa63e0b810c261fe8843e43b8aa7df8f8a5ec161b20411013f6a41b73740fc.  Curious why the tooling missed this dependency.

**Comment by mthalman** [comment]
There are some additional packages that I missed that are being detected as prebuilts. I'll be adding them to this PR:

Microsoft.CodeAnalysis.Analyzers.2.6.1
Microsoft.CodeAnalysis.Common.2.9.0
Microsoft.CodeAnalysis.CSharp.2.9.0
System.Diagnostics.StackTrace.4.3.0
System.Private.Uri.4.3.2
System.Text.Encoding.CodePages.4.3.0
System.ValueTuple.4.3.0
System.Xml.XPath.4.3.0
System.Xml.XPath.XDocument.4.3.0

**Comment by mthalman** [comment]
Ready for review now

File: src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/Microsoft.Extensions.DependencyModel.2.1.0.csproj [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net451;netstandard1.3;netstandard1.6</TargetFrameworks>
```
Comment (MichaelSimons) [inline src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/Microsoft.Extensions.DependencyModel.2.1.0.csproj:4]: Can you remove this net451 so this PR does not cause conflicts with https://github.com/dotnet/source-build-reference-packages/pull/596

File: src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/Microsoft.Extensions.DependencyModel.2.1.0.csproj [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <AssemblyName>Microsoft.Extensions.DependencyModel</AssemblyName>
+    <ProjectTemplateVersion>2</ProjectTemplateVersion>
+    <StrongNameKeyId>MicrosoftAspNetCore</StrongNameKeyId>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net451'">
```
Comment (MichaelSimons) [inline src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/Microsoft.Extensions.DependencyModel.2.1.0.csproj:10]: See preview 451 comment.

File: src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/lib/net451/Microsoft.Extensions.DependencyModel.cs
Comment (MichaelSimons) [inline src/referencePackages/src/microsoft.extensions.dependencymodel/2.1.0/lib/net451/Microsoft.Extensions.DependencyModel.cs:1]: See preview 451 comment.

**Review by MichaelSimons — COMMENTED** [review]
Can you please update the original PR comment to include all packages added.  This will improve the traceability.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**